blockchain/chain_sync/src/sync.rs

```diff
@@ -268,6 +216,68 @@ where
         }
     }
 
+    async fn handle_gossip_block(
+        block: GossipBlock,
+        source: Option<PeerId>,
+        network: SyncNetworkContext<DB>,
+        channel: &Sender<(PeerId, FullTipset)>,
+    ) {
+        let source = match source.clone() {
+            Some(source) => source,
+            None => {
+                warn!("Got a GossipBlock with no Source sender. This should not happen based on Filecoin's GossipSub options");
+                return;
+            }
+        };
+        info!(
+            "Received block over GossipSub: {} from {}",
+            block.header.epoch(),
+            source
+        );
+
+        // Get bls_messages in the store or over Bitswap
+        let bls_messages: Vec<_> = block
+            .bls_messages
+            .into_iter()
+            .map(|m| network.bitswap_get::<UnsignedMessage>(m))
+            .collect();
+
+        let bls_messages = match try_join_all(bls_messages).await {
+            Ok(msgs) => msgs,
+            Err(e) => {
+                warn!("Failed to get UnsignedMessage: {}", e);
+                return;
+            }
+        };
+
+        // Get secp_messages in the store or over Bitswap
+        let secp_messages: Vec<_> = block
+            .secpk_messages
+            .into_iter()
+            .map(|m| network.bitswap_get::<SignedMessage>(m))
+            .collect();
+
+        let secp_messages = match try_join_all(secp_messages).await {
+            Ok(msgs) => msgs,
+            Err(e) => {
+                warn!("Failed to get SignedMessage: {}", e);
+                return;
+            }
+        };
+
+        // Form block
+        let block = Block {
+            header: block.header,
+            bls_messages,
+            secp_messages,
+        };
+        let ts = FullTipset::new(vec![block]).unwrap();
+        channel.send((source, ts)).await;
+        // if self.inform_new_head(source.clone(), &ts).await.is_err() {
+        //     warn!("failed to inform new head from peer {}", source);
+        // }
```

Comment:
Probably worth removing comment and also documenting that this will be informing new head since `channel` isn't very descriptive

Reply:
...oops, this wasn't meant to still be there